quadriga/src/main/java/edu/asu/diging/quadriga/core/service/impl/MappedTripleServiceImpl.java

```diff
@@ -123,15 +156,61 @@ private DefaultMapping toTriple(Predicate predicate) {
 
     private TripleElement toTripleElement(Concept concept) {
         TripleElement tripleElement = new TripleElement();
+
+        tripleElement.setId(concept.getId());
+
         tripleElement.setLabel(concept.getLabel());
         tripleElement.setUri(concept.getUri());
         return tripleElement;
     }
 
     private TripleElement toTripleElement(Predicate predicate) {
         TripleElement tripleElement = new TripleElement();
+
+        tripleElement.setId(predicate.getId());
+
         tripleElement.setLabel(predicate.getLabel());
         tripleElement.setUri(predicate.getRelationship());
         return tripleElement;
     }
+    private String mapConceptUriToDatabaseUri(String mappedTripleGroupId,List<String> equalTo)
+    {
+        List<Concept> concept = conceptRepo.findByMappedTripleGroupId(mappedTripleGroupId);
+        List<String> conceptUris = new ArrayList<>();
+        try {
+            for(Concept eachConcept:concept){
+                conceptUris.add(eachConcept.getUri());
+            }
+            for(String similarUris:equalTo){
+                List<String> listOfUris = processUri(similarUris);
+                listOfUris.retainAll(conceptUris);
+                if(!listOfUris.isEmpty()) {
+                    return listOfUris.get(0);
+                }
+            }
+        }
+        catch(NullPointerException e)
+        {
+            logger.error("Couldn't find concept", e);
+            return "error404Page";
```

Comment:
this is a service class, it should not return controller values (i.e. view names)